modules/collect/src/main/java/com/opengamma/strata/collect/Guavate.java

```diff
@@ -831,6 +832,41 @@ public R next() {
     return collectingAndThen(toImmutableList(), Guavate::combineFuturesAsList);
   }
 
+  //-----------------------------------------------------------------------
+  /**
+   * Converts a list of void futures to a single future.
+   * <p>
+   * Convenience function to avoid manually converting a list to an array.
+   * This is similar to {@link #combineFuturesAsList(List)} but specifically for {@code CompletableFuture<Void>}.
+   * <p>
+   * If any input future completes exceptionally, the result will also complete exceptionally.
+   *
+   * @param futures the void futures, may be empty
+   * @return a void future that combines the input future
+   */
+  public static CompletableFuture<Void> combineFuturesAsVoid(List<? extends CompletableFuture<Void>> futures) {
+    int size = futures.size();
+    CompletableFuture[] futuresArray = futures.toArray(new CompletableFuture[size]);
+    return CompletableFuture.allOf(futuresArray);
```

Comment:
Could this not just be the default implementation? I don't see anything in the signature that requires ImmutableList

Reply:
So change `combineFuturesAsList()` to allow null returned values, and get rid of this?
Could make sense, you'd be building up a list of `null` simply to discard it, but that's not the end of the world.

Reply:
Could that reasoning apply to changing `combineFuturesAsMap()`? Not sure it translates as well, having null values in a map I guess is a kind of failure, and if you'd want to wait on the results you could simply collect the values using the `combineFuturesAsList()`